tracer/src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.NetCore.cs

```diff
@@ -132,9 +140,9 @@ private static void SetDatadogAssembly(string path, Assembly cachedAssembly)
         private readonly struct CachedAssembly
         {
             public readonly string Path;
-            public readonly Assembly Assembly;
+            public readonly Assembly? Assembly;
```

Comment:
I don't think this should ever be null? 🤔 
```suggestion
            public readonly Assembly Assembly;
```

Reply:
line 50 we do: `assemblies.Add(new CachedAssembly(file, null)); `
That said, I haven't spent a second looking into if it made sense.

Reply:
Oh yeah, urgh. I guess nullable enabling in this case really is a complete waste of characters 😂

Reply:
Yeah, it made it harder to review this PR since there's a ton of changes that are just noise :(